kuiper-game/MainRegistry.cs

```diff
@@ -12,10 +14,24 @@ public class MainRegistry : ServiceRegistry
         public MainRegistry()
         {
             this.AddHostedService<MainLoopWorker>();
-            this.AddSingleton<ISolarSystemRepository>(x => new JsonFileSolarSystemRepository(new FileInfo("Data\\Sol.solarsystem")));
+            this.AddSingleton<ICommandParser, ConsoleCommandParser>();
+            Scan((_) =>
+            {
+                _.AssemblyContainingType<IConsoleCommand>();
+                _.AddAllTypesOf<IConsoleCommand>(ServiceLifetime.Singleton);
+            });
+            this.AddSingleton<ISolarSystemRepository>(x => new JsonFileSolarSystemRepository(new FileInfo("Data/Sol.solarsystem")));
             this.AddSingleton<ISolarSystemService, SolarSystemService>();
+            this.AddSingleton<IShipService, ShipService>();
             this.AddSingleton<ICaptainService, CaptainService>();
             this.AddSingleton<ICaptainsConsole, CaptainsConsole>();
+            Scan((_) =>
+            {
+                _.AssemblyContainingType<IEvent>();
+                _.AddAllTypesOf<IEvent>(ServiceLifetime.Singleton);
+            });
```

Comment:
I hoped i could use a similiar approach to inject dependencies into events, but so far no luck